tests/test_manager.py

```diff
@@ -172,6 +172,40 @@ def test_projectmanager_show(mock_project):
         assert mock_manager_projects.called
 
 
+def test_projectmanager_projects_list_args():
+    """
+    Are groups.list() and group.projects.list() called with correct arguments?
+
+    1. Make `api.groups.list` return a list with a single item (a group)
+    2. Verify the call to `group.projects.list` uses the right arguments
+    """
+    mock_group = Mock()
+    mock_groups_list = Mock(return_value=[mock_group])
+    mock_projects_list = Mock(return_value=[])
+    mock_group.projects.list = mock_projects_list
+    mock_api = Mock()
+    mock_api.groups.list = mock_groups_list
```

Comment:
I'm not very happy with the convoluted test setup here. It's so complicated that it would deserve a test on its own to make sure the code actually does what it's supposed to do! :zany_face: :worried:  (Hence the 2-step comment above, note!)

I would have rather preferred to to that with a set of annotations on the test function, to avoid distraction inside the function. Similar to the MR manager show function below.

Do you see a way to make this more elegant? – If not, we can merge this PR as is. It's probably good to go.

Reply:
I grouped the mocks setup code and spent some time coming up with two precise comments for the two resulting blocks.

Probably one of the rare cases where comments make sense. :smirk: If a better solution in code is hard to find.